FuelLabs/fuel-indexer
# feat: add native execution

For more context be sure to read over https://github.com/FuelLabs/fuel-indexer/issues/389

### Description
- PR does a few things:
  - Adds support for native execution
  - Cleans up the API (swapping `expect` and `panic` for `?`)
  - Adds a `hello-world-native` example
    - Is the exact same as the `hello-world` example, but with native execution
  - Integrates native execution into the `forc index` plugin (e.g., `forc index new my-index --native`)
  - Updates stop index functionality to remove indices from the database as well
  - Adds some convenience index utility functions (e.g., `bytes32_from_inputs`) 
    - Users on Discord have complained that transforming types is a PITA so that's what these functions are for
    - In the future we might break these out into a separate crate or something 
 
### Testing steps
- [ ] Pull down the `hello-index-native` example and:
  - Run a fuel node `cargo run --bin fuel-node`
  - Run the example `cargo run --bin hello-index-native -- --manifest examples/hello-world-native/hello_index_native.manifest.yaml`
  - Trigger an event `cargo run --bin hello-bin`
  - Event should be indexed
- [ ] Other functionality should be unaffected
  - Should be able to upload/deploy indices
  - Should be able to stop indices
